engine/apps/auth_token/auth.py

```diff
@@ -149,16 +149,17 @@ def _get_user(request: Request, organization: Organization) -> User:
         except (ValueError, TypeError):
             raise exceptions.AuthenticationFailed("Grafana context must be JSON dict.")
 
-        if "UserId" not in context and "UserID" not in context:
-            raise exceptions.AuthenticationFailed("Invalid Grafana context.")
-
-        try:
-            user_id = context["UserId"]
-        except KeyError:
-            user_id = context["UserID"]
-
         try:
-            return organization.users.get(user_id=user_id)
+            if "UserId" in context or "UserID" in context:
+                try:
+                    user_id = context["UserId"]
+                except KeyError:
+                    user_id = context["UserID"]
+                return organization.users.get(user_id=user_id)
+            elif "Login" in context:
+                return organization.users.get(username=context["Login"])
+            else:
+                raise exceptions.AuthenticationFailed("Grafana context must specify a User or UserID.")
```

Comment:
Would this be equivalent to the above?
```
user_id = context.get("UserId", context.get("UserID"))
if user_id is not None:
    return organization.users.get(user_id=user_id)
elif "Login" in context:
    ...
```

Reply:
I'm not sure if the Login part will stay now that we have ID available from the backend pluigin.